src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,142 @@
+    }
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        testUser = new User();
+    }
+
+    @Test
+    void register_userIsNull_notOk() {
+        testUser = null;
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        testUser.setLogin(null);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_userAlreadyExist_notOk() {
+        testUser.setLogin(USED_LOGIN);
+        storageDao.add(testUser);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        testUser.setAge(null);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        testUser.setAge(INVALID_AGE);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_youngUser_notOk() {
+        testUser.setAge(YOUNG_AGE);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        testUser.setPassword(null);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_lessCharPassword_notOk() {
+        testUser.setPassword(INVALID_PASSWORD);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_failedRegistrationWithValidLogin_notOk() {
+        testUser.setLogin(UNIQUE_LOGIN);
+        testUser.setAge(YOUNG_AGE);
+        testUser.setPassword(null);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_failedRegistrationWithValidAge_notOk() {
+        testUser.setLogin(UNIQUE_LOGIN);
+        testUser.setAge(ADULT_AGE);
+        testUser.setPassword(null);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_failedRegistrationWithValidPassword_notOk() {
+        testUser.setLogin(USED_LOGIN);
+        testUser.setAge(null);
+        testUser.setPassword(VALID_PASSWORD);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_failedRegistration_notOk() {
+        testUser.setLogin(USED_LOGIN);
+        testUser.setAge(INVALID_AGE);
+        testUser.setPassword(INVALID_PASSWORD);
+        assertThrows(UserRegistrationException.class, () -> registrationService.register(testUser));
+    }
```

Comment:
duplicate of `register_nullAge_notOk()`
And this is not a used login, should be a unique one, since you are not putting another user in DB